File: JavaScript/InsertionSort.js
const insertionSort = (arr = []) => {
    const len = arr.length;
    for (let i = 0; i < len; i++) {
      let el = arr[i];
      let j;
      for (j = i - 1; j >= 0 && arr[j] > el; j--)
        arr[j + 1] = arr[j];
      arr[j + 1] = el;
    }
    return arr;
  }

//Example Usage:
// let myArray = [2, 4, 1, 6, -7, 8, 5, 9, 3, 4];
// console.log(insertionSort(myArray));

File: Python/InsertionSort.py
"""Python program for implementation of Insertion Sort"""


def insertion_sort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1

        while j >= 0 and key < arr[j]:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = key


# Driver code to test above
if __name__ == "__main__":
    nums = [12, 11, 13, 5, 6]
    insertion_sort(nums)
    print(nums)
